### PR TITLE
Provider v2

### PR DESCRIPTION
**Pull request by rrousselGit**

### CODE REVIEW

File: docs/src/guides/organizing-stores.mdx [hunk 1]
```diff
@@ -178,8 +176,8 @@ class CounterListPage extends StatelessWidget {
 
 There are multiple types of `Providers` available in the [provider][1] package. The ones most relevant and useful to MobX are:
```
Comment (pavanpodila) [inline docs/src/guides/organizing-stores.mdx]: I think we should mention that it should be at least version 2.0 of provider.

File: docs/src/guides/organizing-stores.mdx [hunk 2]
```diff
@@ -134,12 +134,10 @@ Using the [provider][1] package on pub, you can setup a repository of stores at
 In the example below, we are setting up a store at the app-level:
 
 ```
-final MultiCounterStore store = MultiCounterStore();
-
 class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) => Provider<MultiCounterStore>(
-      value: store,
+      builder: (_) => MultiCounterStore(),
```
Comment (pavanpodila) [inline docs/src/guides/organizing-stores.mdx]: Also I am thinking of going for a `MultiProvider`, as each example may have something to "provide" to others deeper in the example widget tree

**Comment by pavanpodila** [comment]
re-opening to see if CircleCI responds

File: docs/src/guides/organizing-stores.mdx (hunk 2 above)
Comment (rrousselGit) [reply docs/src/guides/organizing-stores.mdx]: Was about to do that but I realized: where is coming from the `examples` variable used on `routes` argument of `MaterialApp`?

**Pull request merged**